### DISCUSSION TITLE
Feature Request to add commit history for Fetched commits in GitHub Desktop

### DISCUSSION THREAD

**Discussion by jason-yak** [community/community#144163] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I'm just in the process of switching from Source Tree to GitHub Desktop and in general I'm preferring how much simpler it is. There's just one thing that I feel is missing. 

I really wish I could see the commit history for fetched commits that have not been pulled yet. This is something Source Tree does, and it's extremely handy to be able to see who the commits are from and review the changes before running Pull. At least I've not found a way to do this. I realise Pull Requests could help fill this role, but our team doesn't use branches much and at some point you'd still have this issue of being blind on what a Pull is actually bringing in.

At the moment without any way to review the fetched commits it's somewhat pointless to even have a separate Fetch action. I do see the number of Pull commits appears after running a fetch, but that doesn't really provide any useful information other than that there are commits there. I would have to jump over to the github.com website in order to review the commits.

Once a fetch has run, could the commits appear in history in a visually different state like they're dimmed out? thanks.

Jason

**Comment by VeerammalV** [comment]
Currently, GitHub Desktop doesn’t display fetched but unpulled commits in the history. This feature has been requested by users who need a way to review upcoming changes before pulling them. In the meantime, as a workaround, you can view the remote repository on github.com to check the commits, but native support for this in GitHub Desktop would be ideal for better visibility and review of pending changes

**Reply by jason-yak** [reply]
Thank you for the reply, glad to know others have requested this as well. It would be a really great UX improvement.
